### ISSUE TITLE
Document compilation process for Windows (x86-x64) 

### ISSUE THREAD

**Issue by hidiego** [fluent/fluent-bit#2626]
**Is your feature request related to a problem? Please describe.**
There is no documentation on the compilation process for Windows x86 and cmake generates solution with only Release x64 | Debug x64 configurations.

**Describe the solution you'd like**
Document process for compilation of Release | Debug x86 libraries

**Comment by fujimotos** [comment]
Actually there's not much to do on Fluent Bit's side.

I documented the complete steps to compile Fluent Bit on Windows here:

https://github.com/fluent/fluent-bit-docs/pull/352/files

To build x86 binary, you just need to perform "Compilation" step with
"x86 Developer command prompt" from Visual Studio.

![Screenshot_2020-10-06_17-39-41](https://user-images.githubusercontent.com/8974561/95178968-fba44f80-07fa-11eb-8067-fee5d17aa04a.png)

In short: it simply depends on which Visual Studio toolkit you choose to use.

**Comment by fujimotos** [comment]
@hidiego Actually there's not much to do on Fluent Bit's side.

> There is no documentation on the compilation process for Windows x86 and cmake generates solution with only Release x64 | Debug x64 configurations.

I documented the complete steps to compile Fluent Bit on Windows
in the following patch:

https://github.com/fluent/fluent-bit-docs/pull/352/files

To build x86 binary, you just need to perform "Compilation" step with
"x86 Developer command prompt" from Visual Studio.

![Screenshot_2020-10-06_17-39-41](https://user-images.githubusercontent.com/8974561/95178968-fba44f80-07fa-11eb-8067-fee5d17aa04a.png)

In short: it simply depends on which Visual Studio toolkit you choose to use.

**Comment by hidiego** [comment]
Alright! I managed to compile fluenbit libraries for Windows32bit architecture.

I followed the guide given by @fujimotos here -> https://github.com/fluent/fluent-bit-docs/pull/352/files also used 'x86 Native Tools Command Prompt for VS 2019'.
I got the following error when executing `cmake .. -G "NMake Makefiles" `  :
[CMakeError.log](https://github.com/fluent/fluent-bit/files/5333670/CMakeError.log) 

I guess I need extra configuration for NMake generator to work properly.
Then I tried the "Visual Studio 16 2019" generator: 
`cmake -G "Visual Studio 16 2019"..`
`cmake --build .`
This worked for x64 Architecture

For anyone wanting to compile the 32bit version of the library, you will have to include the -A option on the cmake command like documented here: https://cmake.org/cmake/help/v3.14/generator/Visual%20Studio%2016%202019.html
So just write:
`cmake -G "Visual Studio 16 2019" -A Win32 ..`
and again
`cmake --build .`

that's it!
Thx @fujimotos for the help! ;)

**Comment by hidiego** [comment]
Alright! I managed to compile fluenbit libraries for Windows32bit architecture.

I followed the guide given by @fujimotos here -> https://github.com/fluent/fluent-bit-docs/pull/352/files also used 'x86 Native Tools Command Prompt for VS 2019'.
I got the following error when executing `cmake .. -G "NMake Makefiles" `  :
[CMakeError.log](https://github.com/fluent/fluent-bit/files/5333670/CMakeError.log) 

I guess I need extra configuration for NMake generator to work properly.
Then I tried the "Visual Studio 16 2019" generator: 
`cmake -G "Visual Studio 16 2019"..`
`cmake --build .`
This worked for x64 Architecture

For anyone wanting to compile the 32bit version of the library, you will have to include the -A option on the cmake command like documented here: https://cmake.org/cmake/help/v3.14/generator/Visual%20Studio%2016%202019.html
So just write:
`cmake -G "Visual Studio 16 2019" -A Win32 ..`
and again
`cmake --build .`

that's it!
Thx @fujimotos for the help! ;)
